clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -107,25 +90,11 @@ public ConfigDef define(String name, Type type, Object defaultValue, Validator v
      * @return This ConfigDef so you can chain calls
      */
     public ConfigDef define(String name, Type type, Object defaultValue, Importance importance, String documentation) {
-        return define(name, type, defaultValue, null, importance, documentation, true);
-    }
-
-    /**
-     * Define a required parameter with no default value
-     *
-     * @param name          The name of the config parameter
-     * @param type          The type of the config
-     * @param validator     A validator to use in checking the correctness of the config
-     * @param importance    The importance of this config: is this something you will likely need to change.
-     * @param documentation The documentation string for the config
-     * @return This ConfigDef so you can chain calls
-     */
-    public ConfigDef define(String name, Type type, Validator validator, Importance importance, String documentation) {
```

Comment:
This was its because it was not used anywhere and its signature collides with the method above (a default value with no validator). 

ex: 
define("name", Type.STRING, **null**, Importance.HIGH, "documentation"
define(String name, Type type, **Object defaultValue**, Importance importance, String documentation)
define(String name, Type type, **Validator validator**, Importance importance, String documentation)

If the functionality is needed in the future using a builder style implementation may be favorable, but I feel this patch is already fairly large for docs jira.